### PR TITLE
Jacoco test coverage configured in project

### PR DESCRIPTION
**Pull request by adam-stasiak**
I configured Jacoco Test Coverage framework for measuring current test code coverage state in project.

In order to run this use:
```mvn -P hadoop2 clean verify```

Then It should be generated html report for all modules in ```../bigdata-interop-coverage/target/coverage-report/html/index.html``` file.

Zrzut ekranu 2019-05-10 o 11 08 36

### CODE REVIEW

File: bigdata-interop-coverage/pom.xml [hunk 1]
```diff
@@ -0,0 +1,148 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
```
Comment (medb) [inline bigdata-interop-coverage/pom.xml]: May you use 2 whitespaces for indentation to be consistent with other pom.xml files in this repo?

File: pom.xml [hunk 2]
```diff
@@ -96,6 +96,7 @@
     <hadoop.three.version>3.2.0</hadoop.three.version>
     <junit.version>4.13-beta-1</junit.version>
     <mockito.version>2.27.0</mockito.version>
+    <jacoco.version>0.8.2</jacoco.version>
```
Comment (medb) [inline pom.xml]: Will it work with 0.8.3 version which seems to be the latest one?

File: pom.xml [hunk 3]
```diff
@@ -104,6 +105,7 @@
     <module>gcsio/pom.xml</module>
     <module>util/pom.xml</module>
     <module>util-hadoop/pom.xml</module>
+    <module>bigdata-interop-coverage/pom.xml</module>
```
Comment (medb) [inline pom.xml]: let's rename this module and folder to just `coverage`

File: pom.xml [hunk 4]
```diff
@@ -235,6 +260,8 @@
               </execution>
             </executions>
           </plugin>
+
```
Comment (medb) [inline pom.xml]: Please remove extra lines

File: pom.xml [hunk 5]
```diff
@@ -441,6 +469,7 @@
       </plugins>
     </pluginManagement>
     <plugins>
+
```
Comment (medb) [inline pom.xml]: Please remove extra lines.

**Comment by adam-stasiak** [comment]
@medb I applied suggestions + fixed small problem with missing coverage results for hadoop3. Coverage task by mistake was placed not globally but only for hadoop2 profile. Now it works.

**Comment by medb** [comment]
Thanks for contribution, I have simplified implementation a little and added `coverage` profile to activate jacoco plugin only when necessary.

**Pull request merged**